src/services/LibraryService.ts

```diff
@@ -51,6 +51,11 @@ export class LibraryService implements Service<fhir4.Library> {
     logger.info(`${req.method} ${req.path}`);
 
     const params = gatherParams(req.query, args.resource);
+    if (req.params.id && params.id) {
+      throw new BadRequestError(
+        'Id argument may not be sourced from both a path parameter and a query or fhir parameter.'
+      );
+    }
```

Comment:
Not a big deal for this task, but maybe we could put this into a function in `validationUtils.ts` since it's used twice? Or we leave it and keep this mind in the future if we need to use it again.